### PR TITLE
Use TypeScript and add Gnome 46 support

### PR DESCRIPTION
**Pull request by ProbstDJakob**
In order to make further development for new contributors easier I've added TypeScript support since it is now officially supported. This will also help migrating the extension to new Gnome versions in the future. Furthermore I checked if it works on Gnome 46 and added it to the `metadata.json`.

Changes made besides adding simple types:
- remove the use of global variables and move them into the `SwitchWorkspacesExtension` class
- use an enum for the constants `UP`and `DOWN` (behaves the same as the numbers with the benefit for restricting the values)
- use a container to encapsulate the warning menu variables since they share the same lifetime and thus only one variable needs to be checked if it is `null`
- use the private visibility modifier instead of underscores
- move the constants of `ConfigurationService` out of the class and make them real constants

### CODE REVIEW

**Comment by ProbstDJakob** [comment]
It would be nice if the issues section would be enabled in order to discuss such changes before implementing them and also this would make it possible to create bug reports and feature requests which other contributors could then fix/implement.

**Comment by cmdillon** [comment]
hi! i noticed you are working on updating this extension. is it ready for gnome 46? really would like to give it a try as the original only works on gnome 45.

**Comment by ProbstDJakob** [comment]
The extension works without any changes to the code with Gnome 46. The only thing which really had to be done is to add Gnome 46 to the `metadata.json`. These changes are primarily meant to provide an easier way to migrate to upcoming Gnome versions which may involve breaking changes to the APIs used.

If you want to use this extension with Gnome 46, you could go to the terminal and enter the following Command. This will skip the version checks and thus enables the (all) extension even though it officially does not support it.

```sh
$ gsettings set org.gnome.shell disable-extension-version-validation true
```

File: simulate-switching-workspaces-on-active-monitor@micheledaros.com/extension.ts [hunk 1]
```diff
@@ -0,0 +1,385 @@
+import Clutter from "gi://Clutter";
```
Comment (micheledaros) [inline simulate-switching-workspaces-on-active-monitor@micheledaros.com/extension.ts:1]: Hi @ProbstDJakob thanks for porting the extension to Typesctipt.
Would you be so kind to update the README with detailed instructions on how to build it?
I've tried on my system (with npm 10.8.1) and got many errors

Comment (ProbstDJakob) [reply simulate-switching-workspaces-on-active-monitor@micheledaros.com/extension.ts:1]: I've created an issue for it: https://github.com/micheledaros/gnome-shell-extension-simulate-switching-workspaces-on-active-monitor/issues/10

**Pull request merged**